code/script.js

```diff
@@ -1,325 +1,824 @@
-}
+  // Generate a board with remaining players
+  generateBoard();
+};
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
-}
+  const confirmation = confirm(`Are you sure you want to guess on ${personToConfirm}`);
+  // only the player confirmed, then checkMyGuess function will be called
+  if (confirmation) {
+    checkMyGuess(personToConfirm);
+  }
+};
 
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  // display needs to be changed to show if the player guesed correctly or not
+  winOrLosePage.style.display = "flex";
+  boardWrapper.style.display = "none";
+
+  // Store final counter number and time for local storage
+  finalCounts = counter;
+  finalTimerValue = timer.textContent;
+
+  // check a name of secret and guessed person's name is the same
+  if (personToCheck === secret.name) {
+    handleLocalStorage();
+
+    winOrLoseText.textContent = `✨🎉Conglaturation!! 🎉✨`;
+    createSound("./images/audio/win-sound.mp3");
+  } else {
+    winOrLoseText.textContent = `💫 Oh no!! A wrong person. 
+                                        Game over 😱 `;
+    createSound("./images/audio/fail-sound.mp3");
+  }
+};
+
+// Handle Local Storage
+
+const handleLocalStorage = () => {
+  const dataStorage = {
+    playerName: player,
+    howManyGuesses: finalCounts,
+    time: finalTimerValue,
+  };
+  // Pushing a newly created obj with game data into the array that stores data from localstorage
+  arrayForLocalStorage.push(dataStorage);
+
+  // then store again in localstorage
+  window.localStorage.setItem("items", JSON.stringify(arrayForLocalStorage));
+};
+
+/*********************************************************************/
+// opening the borad page
+const startGameBtn = document.getElementById("game-start-btn");
+const initialPage = document.getElementById("initial-page");
+
+player = prompt("Who is going to play, today?");
+
+// if player did not type its name, alert and ask again
+while (!player) {
+  alert("Please type your name!");
+  player = prompt("Who is going to play,today");
 }
 
-// Invokes the start function when website is loaded
-start()
+welcomeMessage.innerText = `Welcome to Guess Who, ${player}!! `;
+
+startGameBtn.onclick = function () {
+  initialPage.style.display = "none";
+
+  NameContainer.innerText = `Player: ${player}`;
+  // Invokes the start function when startgameBtn is clicked
+  start();
+};
+
+/*********************************************************************/
+// Sound effect function
```

Comment:
Cool sound effects! I really liked the sound effect when you've guessed wrong :D